File: Java/src/main/java/com/DSA/Java/Algorithms/Sorting/HeapSort/HeapSort.java
package com.DSA.Java.Algorithms.Sorting.HeapSort;

import java.util.Arrays;

public class HeapSort {
  private int[] arr;
  private BinaryHeap bh;

  public HeapSort(int[] arr) {
    bh = new BinaryHeap(arr.length);
    this.arr = arr;
  }

  public void sort() {
    for (int i = 0; i < arr.length; i++) {
      bh.insertValue(arr[i]);
    }

    for (int i = 0; i < arr.length; i++) {
      arr[i] = bh.getTopElement();
    }
  }
  
  @Override
  public String toString() {
    return Arrays.toString(arr);
  }
}

File: C++/Algorithm/src/Sorting/HeapSort.cpp
#include "../../include/Sorting/HeapSort.h"

void HeapSort::sort_array() {
    for (int i = 0; i < arr.size(); i++) {
        bh.insert_element(arr[i]);
    }

    for (int i = 0; i < arr.size(); i++) {
        arr[i] = bh.extract_top_element();
    }

}
